File: frontend/src/api/materialRecognition.ts
import { apiClient, type ApiRequestConfig } from '@/api/client';

export interface PollutantDetail {
  probability: number;
  reason: string;
}

export interface MaterialRecognitionResult {
  material_type: string;
  material_name: string;
  estimated_year_range: string;
  identified_materials: string[];
  likely_pollutants: Record<string, PollutantDetail>;
  confidence_overall: number;
  recommendations: string[];
  description: string;
  has_high_risk: boolean;
}

export const materialRecognitionApi = {
  recognize: async (
    buildingId: string,
    file: File,
    options?: { zoneId?: string; elementId?: string; save?: boolean },
  ): Promise<MaterialRecognitionResult> => {
    const formData = new FormData();
    formData.append('file', file);
    if (options?.zoneId) formData.append('zone_id', options.zoneId);
    if (options?.elementId) formData.append('element_id', options.elementId);
    if (options?.save) formData.append('save', 'true');

    const requestConfig: ApiRequestConfig = {
      headers: { 'Content-Type': 'multipart/form-data' },
      skipRetry: true,
      timeout: 30000,
    };

    const response = await apiClient.post<MaterialRecognitionResult>(
      `/buildings/${buildingId}/materials/recognize`,
      formData,
      requestConfig,
    );
    return response.data;
  },
};

File: backend/app/schemas/material_recognition.py
"""Schemas for material recognition via Claude Vision API."""

from pydantic import BaseModel, ConfigDict, Field


class PollutantDetail(BaseModel):
    probability: float = Field(ge=0.0, le=1.0)
    reason: str = ""


class MaterialRecognitionResult(BaseModel):
    """Result from Claude Vision material recognition."""

    material_type: str
    material_name: str
    estimated_year_range: str = ""
    identified_materials: list[str] = []
    likely_pollutants: dict[str, PollutantDetail] = {}
    confidence_overall: float = Field(ge=0.0, le=1.0)
    recommendations: list[str] = []
    description: str = ""
    has_high_risk: bool = False

    model_config = ConfigDict(from_attributes=True)


class MaterialRecognitionRequest(BaseModel):
    """Optional metadata sent alongside the image upload."""

    building_id: str | None = None
    zone_id: str | None = None
    element_id: str | None = None
    notes: str | None = None
